### ISSUE TITLE
Complete regression test in a MacOS environment

### ISSUE THREAD

**Issue by eliasnogueira** [bonigarcia/webdrivermanager#476]
This is a task to track the regression test results running in a MacOS environment.

### Environment information
* MacOS Catalina version 10.15.4
* Oracle JDK build 1.8.0_241-b07

### Tasks
- [x] Pull request from the latest on master (specify the date)
- [x] Build webdrivermanager
- [x] Run all the test on `src/test/`

### Expectations
All the test should pass, except some tests specifically for Windows environment

#### Browser information
The following browser were installed:
* Google Chrome 81.0.4044.129
* Safari 13.1 (15609.1.20.111.8)
* Firefox 75.0
* Microsoft Edge 81.0.416.68
* Opera 68.0.3618.63

#### Pre conditions
The `.m2/repository/webdriver` was removed

#### Test results
The complete test execution log is attached to
[test_execution.log](https://github.com/bonigarcia/webdrivermanager/files/4580850/test_execution.log)
 this issue.

```
Tests run: 125, Failures: 2, Errors: 2, Skipped: 3
```

##### FirefoxVersionTest
```
2020-05-05 10:58:26 [main] ERROR i.g.bonigarcia.wdm.WebDriverManager.noCandidateUrlFound(601) -- geckodriver (latest version) for MAC 32 not found in https://api.github.com/repos/mozilla/geckodriver/releases
2020-05-05 10:58:26 [main] ERROR i.g.bonigarcia.wdm.WebDriverManager.handleException(719) -- There was an error managing geckodriver (latest version) (geckodriver (latest version) for MAC 32 not found in https://api.github.com/repos/mozilla/geckodriver/releases)
io.github.bonigarcia.wdm.config.WebDriverManagerException: geckodriver (latest version) for MAC 32 not found in https://api.github.com/repos/mozilla/geckodriver/releases
	at io.github.bonigarcia.wdm.WebDriverManager.noCandidateUrlFound(WebDriverManager.java:602)
	at io.github.bonigarcia.wdm.WebDriverManager.createUrlHandler(WebDriverManager.java:751)
	at io.github.bonigarcia.wdm.WebDriverManager.download(WebDriverManager.java:588)
	at io.github.bonigarcia.wdm.WebDriverManager.manage(WebDriverManager.java:520)
	at io.github.bonigarcia.wdm.WebDriverManager.handleException(WebDriverManager.java:717)
	at io.github.bonigarcia.wdm.WebDriverManager.manage(WebDriverManager.java:526)
	at io.github.bonigarcia.wdm.WebDriverManager.setup(WebDriverManager.java:272)
	at io.github.bonigarcia.wdm.test.base.VersionTestParent.testSpecificVersions(VersionTestParent.java:104)
```

```
2020-05-05 10:58:27 [main] ERROR i.g.bonigarcia.wdm.WebDriverManager.noCandidateUrlFound(601) -- geckodriver (latest version) for MAC 64 not found in https://api.github.com/repos/mozilla/geckodriver/releases
2020-05-05 10:58:27 [main] ERROR i.g.bonigarcia.wdm.WebDriverManager.handleException(719) -- There was an error managing geckodriver (latest version) (geckodriver (latest version) for MAC 64 not found in https://api.github.com/repos/mozilla/geckodriver/releases)
io.github.bonigarcia.wdm.config.WebDriverManagerException: geckodriver (latest version) for MAC 64 not found in https://api.github.com/repos/mozilla/geckodriver/releases
	at io.github.bonigarcia.wdm.WebDriverManager.noCandidateUrlFound(WebDriverManager.java:602)
	at io.github.bonigarcia.wdm.WebDriverManager.createUrlHandler(WebDriverManager.java:751)
	at io.github.bonigarcia.wdm.WebDriverManager.download(WebDriverManager.java:588)
	at io.github.bonigarcia.wdm.WebDriverManager.manage(WebDriverManager.java:520)
	at io.github.bonigarcia.wdm.WebDriverManager.handleException(WebDriverManager.java:717)
	at io.github.bonigarcia.wdm.WebDriverManager.manage(WebDriverManager.java:526)
	at io.github.bonigarcia.wdm.WebDriverManager.setup(WebDriverManager.java:272)
	at io.github.bonigarcia.wdm.test.base.VersionTestParent.testSpecificVersions(VersionTestParent.java:104)
```

```
[ERROR] Tests run: 6, Failures: 1, Errors: 2, Skipped: 0, Time elapsed: 4.631 s <<< FAILURE! - in io.github.bonigarcia.wdm.test.firefox.FirefoxVersionTest
[ERROR] testSpecificVersions[0: DEFAULT](io.github.bonigarcia.wdm.test.firefox.FirefoxVersionTest)  Time elapsed: 0.227 s  <<< FAILURE!
java.lang.AssertionError: 

Expected: "0.8.0"
     but: was "0.26.0"

[ERROR] testSpecificVersions[1: 32](io.github.bonigarcia.wdm.test.firefox.FirefoxVersionTest)  Time elapsed: 0.458 s  <<< ERROR!
io.github.bonigarcia.wdm.config.WebDriverManagerException: io.github.bonigarcia.wdm.config.WebDriverManagerException: geckodriver (latest version) for MAC 32 not found in https://api.github.com/repos/mozilla/geckodriver/releases
Caused by: io.github.bonigarcia.wdm.config.WebDriverManagerException: geckodriver (latest version) for MAC 32 not found in https://api.github.com/repos/mozilla/geckodriver/releases

[ERROR] testSpecificVersions[2: 64](io.github.bonigarcia.wdm.test.firefox.FirefoxVersionTest)  Time elapsed: 0.438 s  <<< ERROR!
io.github.bonigarcia.wdm.config.WebDriverManagerException: io.github.bonigarcia.wdm.config.WebDriverManagerException: geckodriver (latest version) for MAC 64 not found in https://api.github.com/repos/mozilla/geckodriver/releases
Caused by: io.github.bonigarcia.wdm.config.WebDriverManagerException: geckodriver (latest version) for MAC 64 not found in https://api.github.com/repos/mozilla/geckodriver/releases
```

##### SeleniumServerStandaloneTest
```
[ERROR] Tests run: 4, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.83 s <<< FAILURE! - in io.github.bonigarcia.wdm.test.other.OtherWebDriverTest
[ERROR] test[0: class org.openqa.selenium.safari.SafariDriver class org.openqa.selenium.WebDriverException](io.github.bonigarcia.wdm.test.other.OtherWebDriverTest)  Time elapsed: 0.812 s  <<< FAILURE!
java.lang.AssertionError: Expected test to throw an instance of org.openqa.selenium.WebDriverException
```

##### VersionsTest
```
2020-05-05 10:59:31 [main] ERROR i.g.bonigarcia.wdm.WebDriverManager.handleException(719) -- There was an error managing chromedriver 81.0.4044.69 (No proper candidate URL to download chromedriver 81.0.4044.69)
io.github.bonigarcia.wdm.config.WebDriverManagerException: No proper candidate URL to download chromedriver 81.0.4044.69
	at io.github.bonigarcia.wdm.online.UrlHandler.getCandidateUrl(UrlHandler.java:284)
	at io.github.bonigarcia.wdm.WebDriverManager.download(WebDriverManager.java:589)
	at io.github.bonigarcia.wdm.WebDriverManager.manage(WebDriverManager.java:520)
	at io.github.bonigarcia.wdm.WebDriverManager.setup(WebDriverManager.java:272)
	at io.github.bonigarcia.wdm.test.versions.IgnoredVersionTest.ignoredVersions(IgnoredVersionTest.java:60)
```

**Comment by eliasnogueira** [comment]
@bonigarcia would you like separate issues for each test failure?

**Comment by bonigarcia** [comment]
No, it's fine like that, thanks a lot.

**Comment by bonigarcia** [comment]
I think all the errors are normal. I mean, no available driver version for giving combinations of the operative system (Mac) and platform. I have just released WebDriverManager 4.0.0. Let's see how this new major version works. Thanks a lot for your help, @eliasnogueira!
